test/metric_naming_test.go

```diff
@@ -0,0 +1,90 @@
+package test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coredns/coredns/plugin/acl"
+	"github.com/coredns/coredns/plugin/autopath"
+	"github.com/coredns/coredns/plugin/cache"
+	"github.com/coredns/coredns/plugin/dns64"
+	"github.com/coredns/coredns/plugin/forward"
+	"github.com/coredns/coredns/plugin/grpc"
+	"github.com/coredns/coredns/plugin/health"
+	"github.com/coredns/coredns/plugin/hosts"
+	"github.com/coredns/coredns/plugin/kubernetes"
+	"github.com/coredns/coredns/plugin/metrics"
+	"github.com/coredns/coredns/plugin/reload"
+	"github.com/coredns/coredns/plugin/template"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/prometheus/util/promlint"
+)
+
+func TestPrometheusMetrics(t *testing.T) {
+	m := metrics.New("")
+	collectors := []prometheus.Collector{
```

Comment:
this is hard to keep to to date and also requires all metric variables are exported which is odd.
A better test is to iterator over the Go code and find the metrics by their type

Reply:
I agree, what do you think of my `package.Metrics` idea?

Reply:
there are already a bunch of unwritten rules about plugin development, i don't want to burden people with adding more. If we (=coredns core dev) care about this we should put in the effort to make this checking possible without burdening plugin developers.

Reply:
Currently,  it used `ast` to parse the file, and build the struct of MetricFamily